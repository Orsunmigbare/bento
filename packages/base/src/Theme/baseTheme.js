import { ThemeManager } from "@staccx/theme"

export const theme = {
  name: "Default theme",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px",
    micro: "3px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px",
    breakout: "640px"
  },
  color: {
    white: "#fff",
    bg: "#fff",
    subtleHover: "#F7F7FF",
    bgGray: "#f7f7ff",
    line: "#EBEBF2",
    gray: "#767676",
    wcag: "#777",
    disabled: "#BABABA",
    primary: "#00a190",
    secondary: "#F5A623",
    text: "#2b2b2b",
    negative: "#ea526f",
    positive: "#4A90E2",
    warning: "#c54059",
    black: "#000",
    green: "#00a190" // for toggle eks
  },
  fontFamily: {
    heading: "sans-serif",
    body: "sans-serif"
  },
  font: {
    h1: "36px",
    h2: "26px",
    h3: "22px",
    h4: "18px",
    h5: "16px",
    h6: "14px",
    base: "14px",
    tiny: "12px",
    input: "16px",
    huge: "44px"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

const manager = new ThemeManager(theme)
export default manager
