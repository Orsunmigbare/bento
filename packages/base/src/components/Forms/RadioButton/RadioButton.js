import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"

const Radio = styled.input`
  position: absolute;
  clip: rect(0, 0, 0, 0);
  clip: rect(0 0 0 0);

  &:checked ~ label::after {
    transform: scale(1);
  }
`

const Label = styled.label`
  position: relative;
  padding: 0 0 0 ${p => p.theme.globals.spacing.mediumPlus};
  width: auto;
  letter-spacing: normal;
  font-family: ${p => p.theme.globals.font.body};
  font-size: ${p => p.theme.globals.font.size.input};
  font-weight: normal;
  cursor: pointer;

  &::before {
    content: "";
    position: absolute;
    top: -4px;
    left: 0;
    display: block;
    width: ${p => p.theme.globals.spacing.medium};
    height: ${p => p.theme.globals.spacing.medium};
    border: 1px solid ${p => p.theme.globals.color.line};
    border-radius: 50%;
    background: ${p => p.theme.globals.color.white};
  }

  &::after {
    content: "";
    position: absolute;
    top: 0px;
    left: 4px;
    display: block;
    width: 16px;
    height: 16px;
    border-radius: 50%;
    background-color: ${p => p.theme.globals.color.primary};
    line-height: 20px;
    transform: scale(0);
    transition: all 0.2s ease-in-out;
  }
`

const RadioWrapper = styled.div`
  min-height: ${p => p.theme.globals.targetSize.normal};
  padding-top: ${p => p.theme.globals.spacing.small};
  padding-bottom: ${p => p.theme.globals.spacing.small};

  &:hover,
  ${Radio}:focus ~ {
    ${Label} {
      &::before {
        border-color: ${p => p.theme.globals.color.wcag};
      }
    }
  }
`

const RadioButton = ({
  children,
  id,
  group,
  disabled,
  onChange,
  className,
  defaultChecked,
  ...otherProps
}) => (
  <RadioWrapper className={className}>
    <Radio
      id={id}
      disabled={disabled}
      type="radio"
      name={group}
      onChange={onChange}
      defaultChecked={defaultChecked}
      {...otherProps}
    />
    <Label htmlFor={id}>{children}</Label>
  </RadioWrapper>
)

RadioButton.propTypes = {
  children: PropTypes.string.isRequired,
  className: PropTypes.string,
  defaultChecked: PropTypes.bool,
  disabled: PropTypes.bool,
  group: PropTypes.string,
  id: PropTypes.string.isRequired,
  input: PropTypes.any,
  onChange: PropTypes.func,
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.object
  ])
}

RadioButton.defaultProps = {
  className: "",
  defaultChecked: false,
  disabled: false,
  input: {},
  onChange: null,
  value: ""
}

export default RadioButton
