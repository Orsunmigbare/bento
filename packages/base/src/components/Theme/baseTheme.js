export const theme = {
  name: "StaccXBase",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px"
  },
  color: {
    bg: "#fff",
    bgGray: "#f7f7ff",
    primary: "#00a190",
    gray: "#767676",
    text: "#2b2b2b",
    white: "#fff",
    red: "#ea526f",
    green: "#00a190",
    blue: "#988EFB",
    yellow: "#f4da87",
    line: "#EBEBF2",
    wcag: "#777",
    disabled: "#BABABA",
    warning: "#c54059",
    black: "#000",
    subtleHover: "#F7F7FF"
  },
  font: {
    size: {
      h1: "36px",
      h2: "26px",
      h3: "22px",
      h4: "18px",
      h5: "16px",
      h6: "14px",
      base: "14px",
      tiny: "12px",
      input: "16px",
      huge: "44px"
    },
    heading: "Libre Baskerville",
    body: "Libre Franklin"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

export default theme
