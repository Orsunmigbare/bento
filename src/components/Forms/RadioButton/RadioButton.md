```jsx
<div>
  <RadioButton id="213ewqs" onChange={() => console.log("Do you copy? Over.")}>
    Radio me
  </RadioButton>
</div>
```
