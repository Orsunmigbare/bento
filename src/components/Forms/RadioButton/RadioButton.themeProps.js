import { themePropTypes, createVariantsFunctionFactory } from "../../../theming"

const themeProps = {
  radio: {
    name: "RADIO_BUTTON_RADIO",
    description: "Radio button style",
    type: themePropTypes.style
  },
  label: {
    name: "RADIO_BUTTON_LABEL",
    description: "Label style",
    type: themePropTypes.style
  },
  wrapper: {
    name: "RADIO_BUTTON_WRAPPER",
    description: "Wrapper style",
    type: themePropTypes.style
  }
}

export default createVariantsFunctionFactory(themeProps)
