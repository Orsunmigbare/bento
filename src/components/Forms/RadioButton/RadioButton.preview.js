import React from "react"
import RadioButton from "./RadioButton"

const preview = {
  title: "RadioButton",
  category: "Components/Forms",
  component: RadioButton,
  render: props => (
    <RadioButton
      id="213ewqs"
      onChange={() => console.log("Do you copy? Over.")}
    >
      Radio me
    </RadioButton>
  )
}

export default preview
