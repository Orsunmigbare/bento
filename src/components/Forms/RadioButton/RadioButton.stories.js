import React from "react"
import { storiesOf } from "@storybook/react"
import RadioButton from "./RadioButton"
import docs from "./RadioButton.md"

storiesOf("Components|Base/Forms/RadioButton")
  .add("no props", () => <RadioButton>RadioButton</RadioButton>, {
    info: { text: docs }
  })
  .add("ID", () => <RadioButton id={"123123"}>Radiobutton</RadioButton>, {
    info: { text: docs }
  })
  .add(
    "log",
    () => (
      <RadioButton
        id={"3424"}
        onChange={e => console.log("Something", e.target.value, e.target.id)}
      >
        {" "}
        RadioButton{" "}
      </RadioButton>
    ),
    {
      info: { text: docs }
    }
  )
  .add(
    "Default checked",
    () => (
      <RadioButton id={"32423423"} defaultChecked>
        Radiobutton
      </RadioButton>
    ),
    {
      info: { text: docs }
    }
  )
