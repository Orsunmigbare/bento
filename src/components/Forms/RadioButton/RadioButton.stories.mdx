import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import RadioButton from "./RadioButton"


<Meta title="Components|Forms/RadioButton" component={RadioButton}/>


# RadioButton
Radio buttons are normally presented in radio groups to describe and select related settings. Only one radio button in a group can be selected at the same time.

# import

```javascript
import {RadioButton} from "@staccx/bento"
```

# Types of RadioButtons


## Standard RadioButton
<Preview>
  <Story name="Standard">
   <RadioButton id="123">RadioButton</RadioButton>
  </Story>
</Preview>

## RadioButtons in a group
<Preview>
  <Story name="Group">
  <>
   <RadioButton id="124" group="test">Radio1</RadioButton>
   <RadioButton id="125" group="test">Radio2</RadioButton>
   <RadioButton id="126" group="test">Radio3</RadioButton>
   </>
  </Story>
</Preview>

## Disabled
if you set the radioButton to disabled it cannot be checked

<Preview>
  <Story name="Disabled">
   <RadioButton id="130" disabled>Disabled RadioButton</RadioButton>
  </Story>
</Preview>

## Default Checked
If you default check a radioButton, it will render as checked

<Preview>
  <Story name="Default Checked">
     <>
      <RadioButton id="134" group="test" defaultChecked>Radio1</RadioButton>
      <RadioButton id="135" group="test">Radio2</RadioButton>
      <RadioButton id="136" group="test">Radio3</RadioButton>
      </>
  </Story>
</Preview>

## Pitfalls
if you need multiple radioButtons in a group, they need the group prop to assign them if not they wil not be assigned together


<Preview>
  <Story name="NoGroup">
  <>
   <RadioButton id="127">Radio1</RadioButton>
   <RadioButton id="128">Radio2</RadioButton>
   <RadioButton id="129">Radio3</RadioButton>
   </>
  </Story>
</Preview>

## If two radioButtons has the same ID they wil only check the first one

<Preview>
  <Story name="sameId">
  <>
   <RadioButton id="120">Radio1</RadioButton>
   <RadioButton id="120">Radio2</RadioButton>
   </>
  </Story>
</Preview>



# Props for RadioButton
<Props of={RadioButton} />
