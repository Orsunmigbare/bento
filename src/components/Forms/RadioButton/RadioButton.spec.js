import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import baseTheme from "../../../theming/themes/baseTheme"
import { render, screen } from "@testing-library/react"
import RadioButton from "./RadioButton"

describe("RadioButton", () => {
  describe("Snapshots", () => {
    it("No props", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <RadioButton>RadioButton</RadioButton>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("ID", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <RadioButton id="ID">RadioButton</RadioButton>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("group", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <RadioButton id="ID" group="ID1">
              RadioButton
            </RadioButton>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("No children", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <RadioButton id="ID" />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})

describe("Rendering", () => {
  it("Should render", () => {
    render(
      <ThemeProvider theme={baseTheme}>
        <RadioButton id="ID" group="ID1">
          RadioButton
        </RadioButton>
      </ThemeProvider>
    )
    expect(screen.getByText("RadioButton")).toBeInTheDocument()
  })
})
