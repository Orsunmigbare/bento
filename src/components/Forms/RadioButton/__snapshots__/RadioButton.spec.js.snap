// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton Snapshots ID 1`] = `
<div
  className="sc-AxiKw deZmix"
>
  <input
    className="sc-AxjAm cxcmXo"
    defaultChecked={false}
    disabled={false}
    id="ID"
    onChange={null}
    type="radio"
    value=""
  />
  <label
    className="sc-AxirZ DKObq"
    htmlFor="ID"
  >
    RadioButton
  </label>
</div>
`;

exports[`RadioButton Snapshots No children 1`] = `
<div
  className="sc-AxiKw deZmix"
>
  <input
    className="sc-AxjAm cxcmXo"
    defaultChecked={false}
    disabled={false}
    id="ID"
    onChange={null}
    type="radio"
    value=""
  />
  <label
    className="sc-AxirZ DKObq"
    htmlFor="ID"
  />
</div>
`;

exports[`RadioButton Snapshots No props 1`] = `
<div
  className="sc-AxiKw deZmix"
>
  <input
    className="sc-AxjAm cxcmXo"
    defaultChecked={false}
    disabled={false}
    onChange={null}
    type="radio"
    value=""
  />
  <label
    className="sc-AxirZ DKObq"
  >
    RadioButton
  </label>
</div>
`;

exports[`RadioButton Snapshots group 1`] = `
<div
  className="sc-AxiKw deZmix"
>
  <input
    className="sc-AxjAm cxcmXo"
    defaultChecked={false}
    disabled={false}
    id="ID"
    name="ID1"
    onChange={null}
    type="radio"
    value=""
  />
  <label
    className="sc-AxirZ DKObq"
    htmlFor="ID"
  >
    RadioButton
  </label>
</div>
`;
